sound/soc/sof/Kconfig

```diff
@@ -186,6 +186,7 @@ endif ## SND_SOC_SOF_DEVELOPER_SUPPORT
 
 config SND_SOC_SOF
 	tristate
+	select VIRTUAL_BUS
```

Comment:
@plbossart please pardon my ignorance here. VIRTUAL_BUS is user visible config option and I guess select was a mistake in the first place. But kconfig is my least favourite subject, so would something like this make better sense?
```
diff --git a/sound/soc/sof/Kconfig b/sound/soc/sof/Kconfig
index 0f8a117502ea..038d94a4ca38 100644
--- a/sound/soc/sof/Kconfig
+++ b/sound/soc/sof/Kconfig
@@ -186,7 +186,6 @@ endif ## SND_SOC_SOF_DEVELOPER_SUPPORT
 
 config SND_SOC_SOF
        tristate
-       select VIRTUAL_BUS
        select SND_SOC_TOPOLOGY
        select SND_SOC_SOF_NOCODEC if SND_SOC_SOF_NOCODEC_SUPPORT
        help
@@ -196,6 +195,13 @@ config SND_SOC_SOF
          module dependencies but since the module or built-in type is decided
          at the top level it doesn't matter.
 
+config SND_SOC_SOF_CLIENT
+       bool
+       depends on VIRTUAL_BUS
+       help
+         This option is not user-selectable but automagically handled by
+         'select' statements at a higher level
+
 config SND_SOC_SOF_PROBE_WORK_QUEUE
        bool
        help
diff --git a/sound/soc/sof/intel/Kconfig b/sound/soc/sof/intel/Kconfig
index 9b3c01072dd8..5ed7ac4c5866 100644
--- a/sound/soc/sof/intel/Kconfig
+++ b/sound/soc/sof/intel/Kconfig
@@ -2,6 +2,7 @@
 config SND_SOC_SOF_INTEL_TOPLEVEL
        bool "SOF support for Intel audio DSPs"
        depends on X86 || COMPILE_TEST
+       select SND_SOC_SOF_CLIENT
        help
          This adds support for Sound Open Firmware for Intel(R) platforms.
          Say Y if you have such a device.
```

Reply:
or should I make SND_SOC_SOF_CLIENT a user visible option?

Reply:
I think yes, that should be an option in my opinion. If you don't want the extra overhead then don't use it - but it'll come with fewer options like no probes, IPC flood, etc.

Reply:
@plbossart with this change, when we make audio a separate client, it will become optional too. Is that ok?